### ISSUE TITLE
Incremental Java compilation is broken with Android 3.5.1 and Gradle 6.0

### ISSUE THREAD

**Issue by wolfs** [gradle/gradle#11330]
See https://issuetracker.google.com/issues/144011039

Changes to sources added by the Android Gradle plugin to `AndroidJavaCompile.getSources()` are ignored for incremental Java compilation in Gradle 6.0. When changing a source file, the output is something like this:
```
> Task :mylibrary1:compileReleaseJavaWithJavac UP-TO-DATE
Task :mylibrary1:compileReleaseJavaWithJavac in app Starting
Build cache key for task ':mylibrary1:compileReleaseJavaWithJavac' is 669111c814f7b237087e6a217a4ff93a
Task ':mylibrary1:compileReleaseJavaWithJavac' is not up-to-date because:
  Input property 'sources' file /home/user/test/mylibrary1/src/main/java/com/test/mylibrary1/LibClass.java has changed.
```

Note that there isn't a mention of the input property `stableSources`, which is actually tracked for changes. Not the property `source` of the `SourceTask` is mentioned.

**Comment by wolfs** [comment]
cc @gradle/build-cache

**Comment by wolfs** [comment]
See https://github.com/gradle/gradle/pull/11359 for the smoke tests.

**Comment by wolfs** [comment]
Fixed via #11331.
